fe/fe-core/src/main/java/com/starrocks/common/MarkedCountDownLatch.java

```diff
@@ -65,8 +65,10 @@ public synchronized void countDownToZero(Status status) {
         if (st.ok()) {
             st = status;
         }
-        while (getCount() > 0) {
+        long count = getCount();
```

Comment:
What's the change meaning?

Reply:
Reduce the number of atomic operations.

Reply:
You don't reduce the number of atomic operations.

```
    private static final class Sync extends AbstractQueuedSynchronizer {
        private static final long serialVersionUID = 4982264981922014374L;

        Sync(int count) {
            setState(count);
        }

        int getCount() {
            return getState();
        }

        protected int tryAcquireShared(int acquires) {
            return (getState() == 0) ? 1 : -1;
        }

        protected boolean tryReleaseShared(int releases) {
            // Decrement count; signal when transition to zero
            for (;;) {
                int c = getState();
                if (c == 0)
                    return false;
                int nextc = c - 1;
                if (compareAndSetState(c, nextc))
                    return nextc == 0;
            }
        }
    }
```
super.countDown() will call `compareAndSetState`,  `getCount ` only directly get the `volatile int state`.

So I think we don't need this change.

Reply:
If I am correct, a non-volatile read is always faster than a volatile read on the X86 architecture.  But since this is not a performance-critical code and has nothing to do with the bug, I'll not submit this change.

Reply:
updated, PTAL @kangkaisen